Add default get*ItemViewType implementations to AbstractTableAdapter

MGaetan89 opened the pull request:
This PR adds default implementation of `getColumnHeaderItemViewType`, `getRowHeaderItemViewType` and `getCellItemViewType` to `AbstractTableAdapter`. This replicates the behaviour of `RecyclerView.Adpter`, where `getItemViewType` doesn't need to be overriden if we only have one type.

MGaetan89 commented:
@evrencoskun any plan to do a new release?
I think some of these changes could be useful: https://github.com/evrencoskun/TableView/compare/v0.8.9.2...master

evrencoskun commented:
Hi, @MGaetan89 
Yes, you are right! 
But I'm facing an issue when I try to generate a new release.  So, firstly, I need to have a look at it.

Cheers!

MGaetan89 commented:
When I updated the plugins and dependencies in #297, I changed the way to generate a new release.
I've documented it in this comment: https://github.com/evrencoskun/TableView/pull/297#discussion_r399788141
Have you tried that? Let me know if I can help you in any way :)

I made that changes because `com.novoda:bintray-release` was not compatible with Gradle 6+ (and hasn't been updated since).

evrencoskun commented:
@MGaetan89 
As far as I remember,  I had read your comments before I tried it. But I'll try one more time. 

Thank you for your collaboration :)

Zardozz commented:
May be it is worth publishing elsewhere if you are having problems as it seem JCenter/Bintray is closing down

https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

evrencoskun commented:
@Zardozz Right - In the long run, it will be necessary to move elsewhere.

Cheers!

Zardozz commented:
I've just published another project to Maven Central (one of the alternatives)

The guide https://getstream.io/blog/publishing-libraries-to-mavencentral-2021/ was very good

evrencoskun commented:
@Zardozz noted - Thanks!

santiihoyos commented:
Same problem

MGaetan89 commented:
What problem?
